packages/logger/src/Logger.ts

```diff
@@ -205,7 +203,14 @@ class Logger extends Utility implements ClassThatLogs {
    * @returns {Logger}
    */
   public createChild(options: ConstructorOptions = {}): Logger {
-    return cloneDeep(this).setOptions(options);
+    const parentsPowertoolsLogData = this.getPowertoolLogData();
+    const childLogger = new Logger(merge({}, parentsPowertoolsLogData, options));
+    const parentsPersistentLogAttributes = this.getPersistentLogAttributes();
+    childLogger.addPersistentLogAttributes(parentsPersistentLogAttributes);
+    const childPowertoolsLogData = childLogger.getPowertoolLogData();
+    childLogger.addToPowertoolLogData(merge({}, parentsPowertoolsLogData, childPowertoolsLogData));
```

Comment:
@shdq: question: why do we merge and use the `parentsPowertoolsLogData` twice? Maybe worth documenting with a comment (inside the method, so we don't expose it in the docstring)